Add blog post about Elevate and CentOS 6

randommen96 opened the pull request:
I still need to replace the header image, currently there is a placeholder from another post.

Open for feedback in any form, about content, readability, grammar, anything you can think of :-)

bennyvasquez commented:
Thanks for submitting this, @randommen96! As discussed in mattermost, I'll review this over the next couple days, and we'll schedule it for the 22nd. So glad to see this!

bennyvasquez commented on content/blog/how-elevate-supports-business-needs.md:
```diff
@@ -0,0 +1,182 @@
+    image: /blog-images/2024/2024-04-23-centos6-to-centos7.png
+---
+
+# How Elevate supports business needs
+While we've been busy upgrading all our CentOS 7 servers to AlmaLinux 8 and higher, the ELevate Project now has the abillity to upgrade CentOS 6 servers too!
```
```suggestion
# How Elevate supports business needs
As long-time users of AlmaLinux, we were already familiar with Project ELevate for upgrading CentOS 7, but just a few weeks ago we got some exciting news. While we've been busy upgrading all our CentOS 7 servers to AlmaLinux 8 and higher, the ELevate Project now has the ability to upgrade CentOS 6 servers too!
```
This just felt like it needed a little more context for readers.

randommen96 replied:
Agreed!

bennyvasquez commented on content/blog/how-elevate-supports-business-needs.md:
```diff
@@ -0,0 +1,182 @@
+
+## whoami and a bit about our systems
+
+I'm David from the Netherlands and I try to spread the feeling I experience myself while working within the hosting industry whenever I can, it really is something magical!
```
```suggestion
I'm David from the Netherlands, and I try to spread the feeling I experience myself while working within the hosting industry whenever I can. It really is something magical!
```

bennyvasquez commented on content/blog/how-elevate-supports-business-needs.md:
```diff
@@ -0,0 +1,182 @@
+The first Linux I ever used was actually Debian, which I luckily still use a lot too, as it is the base for [Proxmox Virtual Environment](https://www.proxmox.com).
+While for the past 4 years CentOS and AlmaLinux took most of my time, Debian always kept a special place because of one simple reason, Debian provides in-place OS upgrades.
+
+Even though servers can, and one can discuss, maybe even should be configured in state so you can destroy and replace it easily, that is not always the case.
```
```suggestion
Even though servers can - and maybe even should - be configured in a state that you can destroy and replace it easily, that is not always the case.
```

bennyvasquez commented on content/blog/how-elevate-supports-business-needs.md:
```diff
@@ -0,0 +1,182 @@
+
+Even though servers can, and one can discuss, maybe even should be configured in state so you can destroy and replace it easily, that is not always the case.
+
+So when AlmaLinux came around and further down the road ELevate popped up offering in-place upgrades from CentOS 7 systems to for example AlmaLinux 8 and higher, it also entered that special place!
```
```suggestion
So, when AlmaLinux came around and further down the road introduced ELevate, offering enterprise linux in-place upgrades from CentOS 7 systems to (for example) AlmaLinux 8 and higher, it also entered that special place!
```

bennyvasquez commented on content/blog/how-elevate-supports-business-needs.md:
```diff
@@ -0,0 +1,182 @@
+So when AlmaLinux came around and further down the road ELevate popped up offering in-place upgrades from CentOS 7 systems to for example AlmaLinux 8 and higher, it also entered that special place!
+
+## Why there can be (and are!) still CentOS 6 servers alive and kicking
+We always used CentOS as our main operating system for our webhosting platforms, while we were eventually forced to look elsewhere, we still had a lot of CentOS 7 and a couple of CentOS 6 systems around,
+since ELevate came around we started upgrading our whole fleet to AlmaLinux 8 as we started using that for our new deployments too, we're almost done doing so.
```
```suggestion
We always used CentOS as our main operating system for our webhosting platforms, while we were eventually forced to look elsewhere, we still had a lot of CentOS 7 and a couple of CentOS 6 systems around. Since ELevate came out, we started upgrading our whole fleet to AlmaLinux 8 (as we are using that for our new deployments too). We're almost done doing so.
```

bennyvasquez commented on content/blog/how-elevate-supports-business-needs.md:
```diff
@@ -0,0 +1,182 @@
+We always used CentOS as our main operating system for our webhosting platforms, while we were eventually forced to look elsewhere, we still had a lot of CentOS 7 and a couple of CentOS 6 systems around,
+since ELevate came around we started upgrading our whole fleet to AlmaLinux 8 as we started using that for our new deployments too, we're almost done doing so.
+
+Upgrading is most of the time the less painfull way to get your customer to use a new OS,
+as with migrating to a new server there's always something that takes some debugging where in-place upgrades keep it mostly intact,
+which is important, because as a techie we tend to forget that customers mostly want the servers to just work to support their business needs.
```
```suggestion
Upgrading is the less painful way to get your customer to use a new OS, most of the time, as migrating to a new server always comes with some debugging. In-place upgrades keep it mostly intact, which is important to customers. As techs, we can sometimes forget that customers mostly want the servers to just work, and to support their business needs.
```

bennyvasquez commented on content/blog/how-elevate-supports-business-needs.md:
```diff
@@ -0,0 +1,182 @@
+as with migrating to a new server there's always something that takes some debugging where in-place upgrades keep it mostly intact,
+which is important, because as a techie we tend to forget that customers mostly want the servers to just work to support their business needs.
+
+That still doesn't make up for the CentOS 6 servers currently around, which in all honesty should've been gone a long time ago already,
+in which as a MSP you can only do so much for your customer,
+if in the end they use an application for their business that for some reason only runs on CentOS 6,
+or their developer left a long time ago, well, good luck with replacing that server...
```
```suggestion
That doesn't excuse the CentOS 6 servers still around, which in all honesty should've been gone a long time ago. As an MSP you can only do so much for your customer. If, in the end, they use an application for their business that for some reason only runs on CentOS 6, or their developer left a long time ago, well, good luck with replacing that server.
```

bennyvasquez commented on content/blog/how-elevate-supports-business-needs.md:
```diff
@@ -0,0 +1,182 @@
+if in the end they use an application for their business that for some reason only runs on CentOS 6,
+or their developer left a long time ago, well, good luck with replacing that server...
+
+It gives however a great reason to have some fun with the new release of [ELevate](https://almalinux.org/blog/2024-04-25-elevate-supports-centos-6-to-centos-7/), as it now supports ELevating CentOS 6 too!
```
```suggestion
It gives us a great reason to have some fun with the new release of [ELevate](https://almalinux.org/blog/2024-04-25-elevate-supports-centos-6-to-centos-7/), as it now supports ELevating CentOS 6 too!
```

bennyvasquez commented on content/blog/how-elevate-supports-business-needs.md:
```diff
@@ -0,0 +1,182 @@
+
+It gives however a great reason to have some fun with the new release of [ELevate](https://almalinux.org/blog/2024-04-25-elevate-supports-centos-6-to-centos-7/), as it now supports ELevating CentOS 6 too!
+
+Nowadays there are less and less excuses to make, as [DirectAdmin](https://www.directadmin.com) for example made it even possible to run php 5.6 code on AlmaLinux 9,
+which is a great way of offering your customer a up-to-date operating system if they're really dependent on php 5.6, which gives us one less thing to worry about.
```
```suggestion
Nowadays there are fewer and fewer excuses to prevent upgrades, as [DirectAdmin](https://www.directadmin.com) for example made it even possible to run PHP 5.6 code on AlmaLinux 9. It's a great way of offering your customers an up-to-date operating system if they're really dependent on PHP 5.6. It gives us one less thing to worry about.
```

bennyvasquez commented on content/blog/how-elevate-supports-business-needs.md:
```diff
@@ -0,0 +1,182 @@
+Nowadays there are less and less excuses to make, as [DirectAdmin](https://www.directadmin.com) for example made it even possible to run php 5.6 code on AlmaLinux 9,
+which is a great way of offering your customer a up-to-date operating system if they're really dependent on php 5.6, which gives us one less thing to worry about.
+
+So when CloudFest and AlmaLinux day came around great things happened at the hackathon and the idea for ELevating CentOS 6 servers to CentOS 7 was born, which is really exciting news.
+
+Once benny posted that ELevate was available for CentOS 6 I was really excited to try it on one of our servers, so after a bit of cloning and snapshotting off we went!
```
```suggestion
So when CloudFest and [AlmaLinux Day](https://almalinux.org/blog/2024-05-01-almalinux-day-germany-recap/) came around, great things happened at the hackathon and the idea for ELevating CentOS 6 servers to CentOS 7 was born. Once benny posted that ELevate was available for CentOS 6 I was excited to try it on one of our servers, so after a bit of cloning and snapshotting off we went!
```

bennyvasquez commented on content/blog/how-elevate-supports-business-needs.md:
```diff
@@ -0,0 +1,182 @@
+
+Things that interest me a lot are automation, mostly anything opensource, networking and infrastructure.
+
+The first Linux I ever used was actually Debian, which I luckily still use a lot too, as it is the base for [Proxmox Virtual Environment](https://www.proxmox.com).
+While for the past 4 years CentOS and AlmaLinux took most of my time, Debian always kept a special place because of one simple reason, Debian provides in-place OS upgrades.
```
```suggestion
The first Linux I ever used was actually Debian, which I luckily still use a lot too, as it is the base for [Proxmox Virtual Environment](https://www.proxmox.com). While for the past 4 years CentOS and AlmaLinux took most of my time, Debian always kept a special place for one simple reason, Debian provides in-place OS upgrades.
```

bennyvasquez commented on content/blog/how-elevate-supports-business-needs.md:
```diff
@@ -0,0 +1,182 @@
+One that bothered us the most was that the /boot partition is almost always in need of more space,
+while in more recent installations it was a matter of cleaning up old kernels, in our CentOS 6 servers that really wasn't happening as most servers had a 128Mb /boot partition.
+
+So I had to move the /boot partition to a new partition which was bigger, again, something to keep in mind,
+while ELevate is a great tool to help deprecating those last CentOS 6 servers, it still might be cleaner to do it from scratch some day.
```
```suggestion
So I had to move the /boot partition to a new partition which was bigger. While ELevate is a great tool to help deprecate those last CentOS 6 servers, it still might be cleaner to do it from scratch someday.
```

bennyvasquez commented on content/blog/how-elevate-supports-business-needs.md:
```diff
@@ -0,0 +1,182 @@
+So I had to move the /boot partition to a new partition which was bigger, again, something to keep in mind,
+while ELevate is a great tool to help deprecating those last CentOS 6 servers, it still might be cleaner to do it from scratch some day.
+
+While moving the partition is quite an easy way, it can be quite dirty if you put it directly behind your data partition,
+you might want to consider moving that further back and just expanding the current /boot partition itself,
+or even simpler, move it to a seperate disk, for which the steps are as follows:
```
```suggestion
While moving the partition is quite an easy way, it can be quite dirty if you put it directly behind your data partition. You might want to consider just expanding the current /boot partition itself, or even simpler, move it to a seperate disk, for which the steps are as follows:
```

bennyvasquez commented on content/blog/how-elevate-supports-business-needs.md:
````diff
@@ -0,0 +1,182 @@
+reboot now
+```
+
+### grub to grub2!
+When upgrading to CentOS 7 my test server switches from grub to grub2, when I wanted to try ELevating it to AlmaLinux 8 it complained about some grub issues,
+I had to create a /boot/grub2/grub.cfg file and create /etc/default/grub.
````
```suggestion
### grub to grub2!
After upgrading to CentOS 7, my test server switched from grub to grub2. When I wanted to ELevate it to AlmaLinux 8 it complained about some grub issues. I had to create a /boot/grub2/grub.cfg file and create /etc/default/grub.
```

bennyvasquez commented on content/blog/how-elevate-supports-business-needs.md:
```diff
@@ -0,0 +1,182 @@
+When upgrading to CentOS 7 my test server switches from grub to grub2, when I wanted to try ELevating it to AlmaLinux 8 it complained about some grub issues,
+I had to create a /boot/grub2/grub.cfg file and create /etc/default/grub.
+
+For /etc/default/grub I created it by using the booted system and put the following there:
```
```suggestion
For /etc/default/grub I created it by using the booted system and put the following in the file:
```

bennyvasquez commented on content/blog/how-elevate-supports-business-needs.md:
````diff
@@ -0,0 +1,182 @@
+GRUB_THEME="/boot/grub2/themes/system/theme.txt"
+```
+
+And generate /boot/grub2/grub.cfg with:
````
```suggestion
Then, I generated /boot/grub2/grub.cfg with:
```

bennyvasquez commented on content/blog/how-elevate-supports-business-needs.md:
````diff
@@ -0,0 +1,182 @@
+```
+
+### DirectAdmin specific tips
+After every leapp upgrade we rebuild all packages that DirectAdmin brings, sometimes this can go wrong with for example some older local libraries, the following commands might help you out sometime.
````
```suggestion
After every `leapp upgrade,` we rebuild all packages that DirectAdmin includes. Sometimes this can go wrong, for example with some older local libraries. The following commands might help you out, too.
```

bennyvasquez commented on content/blog/how-elevate-supports-business-needs.md:
````diff
@@ -0,0 +1,182 @@
+da build doMigrateToSystemCurl
+```
+
+Something to keep in mind is in general on CentOS 6 you're running older software versions, as you must for example update MySQL too from 5.6 to 8.0 in the process, you will run into changed or removed configuration parameters,
+change or comment as needed in /etc/my.cnf to get you running again.
````
```suggestion
Something to keep in mind is in general - on CentOS 6 you're running older software versions. That means, you must, for example, update MySQL too from 5.6 to 8.0 in the process of upgrading the server, and you will run into changed or removed configuration parameters. You can change or comment as needed in /etc/my.cnf to get MySQL running again.
```

bennyvasquez commented on content/blog/how-elevate-supports-business-needs.md:
````diff
@@ -0,0 +1,182 @@
+query_cache_size=32M
+```
+
+### Test, test and... test.
+Something easily forgotten, and to be really honest, we've all done it at one point or another by thinking: "easy, I've done that so many times, no way it goes wrong!",
+is to always define some testing points and write down / share the results before and after the change, in this case upgrading.
````
```suggestion
### Test, test and... test.
Something easily forgotten is to define some testing points, and write down and share the results before and after the upgrade. To be honest, we've all done it at one point or another by thinking: "easy, I've done that so many times, no way it goes wrong!".
```

bennyvasquez commented on content/blog/how-elevate-supports-business-needs.md:
```diff
@@ -0,0 +1,182 @@
+Something easily forgotten, and to be really honest, we've all done it at one point or another by thinking: "easy, I've done that so many times, no way it goes wrong!",
+is to always define some testing points and write down / share the results before and after the change, in this case upgrading.
+
+It also gives a really good starting point when cleaning up older packages afterwards, sometimes the server has lived quite long and not every package installed is still needed,
+when you know what to expect from the services your customer provides on their server you can actually clean up packages and test afterwards if all customer critical services are still working.
```
```suggestion
ELevate also gives a good starting point when cleaning up older packages afterward. Sometimes the server has lived quite long and not every package installed is still needed. When you know what to expect from the services your customer provides on their server, you can clean up packages and test afterward to ensure all customer critical services are still working.
```

bennyvasquez commented on content/blog/how-elevate-supports-business-needs.md:
```diff
@@ -0,0 +1,182 @@
+
+
+## AlmaLinux and ELevate helps us sleep at night
+Overall AlmaLinux offers us a really stable platform which is exactly what we want and need,
+one of their slogans is "No drama, just Linux", which is true if you follow the different developments in Linux Land,
+the way they adapted to the several changes Red Hat brought us was really comforting and I think a great way forward.
```
```suggestion
Overall, AlmaLinux offers us a stable platform, which is exactly what we want and need. One of their slogans is "No drama, just Linux", which is true if you follow the different developments in Linux Land. The way they adapted to the several changes Red Hat brought us was very comforting, and I think a great way forward.
```

bennyvasquez commented on content/blog/how-elevate-supports-business-needs.md:
```diff
@@ -0,0 +1,182 @@
+
+The fact that they are able to patch CVE's themselves now as they see fit and even share their own [Errata](https://errata.almalinux.org/) is awesome!
+
+ELevate helps us keeping servers up-to-date and secure for another decade while providing the least impact for our customers,
+which in the end is what counts and makes ourselves and our customers really happy.
```
```suggestion
ELevate helps us keep servers up-to-date and secure for another decade, while providing the least impact for our customers. In the end is what counts and makes us and our customers really happy.
```

bennyvasquez commented on content/blog/how-elevate-supports-business-needs.md:
```diff
@@ -0,0 +1,182 @@
+ELevate helps us keeping servers up-to-date and secure for another decade while providing the least impact for our customers,
+which in the end is what counts and makes ourselves and our customers really happy.
+
+If you have any questions after reading the above story feel free contact me at any possible way, one of them is via [AlmaLinux Chat](https://chat.almalinux.org), thanks for reading!
```
```suggestion
If you have any questions after reading the above story feel free to contact me. The easiest way is via [AlmaLinux Chat](https://chat.almalinux.org). Thanks for reading!
```

bennyvasquez reviewed (commented):
It's so great to see such a great first-hand account of your use of ELevate! All of my suggestions are either grammatical or formatting. Somehow you ended up with a lot of linebreaks in the middle of sentences or paragraphs, so I made a LOT of suggestions to fix those. In a couple places the way you phrased something made it clear to me what you were saying, but was not good english phrasing, and wouldn't have translated well, so I suggested a re-phrase. If I misunderstood what you were saying, feel free to say! 

I also made a header image for you, finally! 

![elevatebusinessneeds](https://github.com/AlmaLinux/almalinux.org/assets/13630986/19f3e4e0-4c33-4046-87af-62e3920ce693)

randommen96 commented:
Thanks benny! I think it is what I get for using nano to write the blog... :-)
I reviewed the corrections which were - much needed - as now the post is readable a lot better.

I added the header image, did one more formatting change and added a reason why we chose to move the /boot partition instead of resizing all partitions.

I think it is complete this way!

bennyvasquez reviewed (approved):
Thanks, @randommen96! I'll push the buttons to merge this in tomorrow morning eastern US time!
